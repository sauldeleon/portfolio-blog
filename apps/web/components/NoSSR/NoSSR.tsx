import dynamic from 'next/dynamic'
import React from 'react'

function NoSsrComponent({ children }: { children: React.ReactNode }) {
  return <>{children}</>
}

export const NoSSR = dynamic(() => Promise.resolve(NoSsrComponent), {
  ssr: false,
})
