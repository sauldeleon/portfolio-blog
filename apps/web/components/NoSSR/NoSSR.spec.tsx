import { render } from '@testing-library/react'

import { NoSSR } from './NoSSR'

jest.mock('next/dynamic', () => ({
  __esModule: true,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  default: (...props: any) => {
    const dynamicModule = jest.requireActual('next/dynamic')
    const dynamicActualComp = dynamicModule.default
    const RequiredComponent = dynamicActualComp(props[0])
    RequiredComponent.preload
      ? RequiredComponent.preload()
      : RequiredComponent.render.preload()
    return RequiredComponent
  },
}))

describe('NoSSR', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<NoSSR>test</NoSSR>)
    expect(baseElement).toMatchSnapshot()
  })
})
