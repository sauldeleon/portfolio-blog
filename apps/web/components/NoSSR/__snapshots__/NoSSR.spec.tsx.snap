// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoSSR should render successfully 1`] = `
<body>
  <div>
    test
  </div>
</body>
`;
